src/test/unit/resources/test_detectors.py

```diff
@@ -0,0 +1,50 @@
+import json
+import os
+
+import pytest
+from opentelemetry.sdk import resources
+
+from lumigo_opentelemetry.resources.detectors import (
+    ProcessResourceDetector,
+    LumigoDistroDetector,
+    EnvVarsDetector,
+)
+
+
+@pytest.fixture(autouse=True)
+def keep_env_vars():
```

Comment:
@nirLumigo I'd like to see a test for `OTEL_SERVICE_NAME` and one for an invalid value of `OTEL_RESOURCE_ATTRIBUTES`

Reply:
Take a look at `assert_otel_detector` in test_boto3.